src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -267,25 +271,81 @@ public void getDocIds(DocIdPusher pusher) throws IOException,
    * Adds all specified metadata columns to the record or response being built.
    */
   private void addMetadata(MetadataHandler meta, ResultSet rs)
-      throws SQLException {
+      throws SQLException, IOException {
     ResultSetMetaData rsMetaData = rs.getMetaData();
     int numberOfColumns = rsMetaData.getColumnCount();
     for (int i = 1; i < (numberOfColumns + 1); i++) {
       String columnName = rsMetaData.getColumnLabel(i);
+      int columnType = rsMetaData.getColumnType(i);
       log.log(Level.FINEST, "Column name: {0}, Type: {1}", new Object[] {
-          columnName, rsMetaData.getColumnType(i)});
-
-      Object value = rs.getObject(i);
+          columnName, columnType});
+
+      Object value = null;
+      switch (columnType) {
+        case Types.CLOB:
+          Clob clob = rs.getClob(i);
+          if (clob != null) {
+            try {
+              value = clob.getSubString(1, 4096);
+            } finally {
+              try {
+                clob.free();
+              } catch (Exception e) {
+                log.log(Level.FINEST, "Error closing CLOB", e);
+              }
+            }
+          }
+          break;
+        case Types.NCLOB:
+          NClob nclob = rs.getNClob(i);
+          if (nclob != null) {
+            try {
+              value = nclob.getSubString(1, 4096);
+            } finally {
+              try {
+                nclob.free();
+              } catch (Exception e) {
+                log.log(Level.FINEST, "Error closing NCLOB", e);
+              }
+            }
+          }
+          break;
+        case Types.LONGVARCHAR:
+          try (Reader reader = rs.getCharacterStream(i)) {
+            if (reader != null) {
+              char[] buffer = new char[4096];
+              int len;
+              if ((len = reader.read(buffer)) != -1) {
+                value = new String(buffer, 0, len);
+              }
+            }
+          }
+          break;
+        case Types.LONGNVARCHAR:
+          try (Reader reader = rs.getNCharacterStream(i)) {
+            if (reader != null) {
+              char[] buffer = new char[4096];
+              int len;
+              if ((len = reader.read(buffer)) != -1) {
+                value = new String(buffer, 0, len);
+              }
+            }
+          }
+          break;
+        default:
+          value = rs.getObject(i);
+          break;
+      }
       String key = metadataColumns.getMetadataName(columnName);
-      if (key != null) {
+      if (key != null && value != null) {
```

Comment:
This is a behavioral change, from indexing "null" to not indexing this metadata value. Let's not do that here as part of this change.